### PR TITLE
Update csvs

### PR DESCRIPTION
**Pull request by clareconstantine**
Quick/list view and cohort view now generate CSVs with the **same columns**. The cohort view CSV still only includes applications that have been accepted, while the quick view CSV includes applications of any status.

[spec](https://docs.google.com/document/d/1RqmmoD0W7I4BFvlwL7mcfxdDW86XgvK-GLpxNVkvpaQ/edit)

sample quick view csv for csp teacher applications (converted to pdf bc github doesn't understand CSVs): [csp_teachers_applications.pdf](https://github.com/code-dot-org/code-dot-org/files/2527506/csp_teachers_applications.61.pdf)

cc @tanyaparker

### CODE REVIEW

**Comment by tanyaparker** [comment]
- Rename "My school is a" to "School Type"
- Rename "Gender identity" to "Teacher's gender identity"
- Rename "Race" to "Teacher's race"

And we just talked about

- Add "Registered for workshop?" after "Assigned Workshop"
- Add "Principal Approval Form URL" after "How did you hear about this program? (Teacher’s response)"

**Comment by tanyaparker** [comment]
[Fix application dashboard CSVs](https://trello.com/c/9tR4E9wD/751-fix-application-dashboard-csvs)

File: dashboard/app/controllers/api/v1/pd/applications_controller.rb [hunk 1]
```diff
@@ -276,9 +269,14 @@ def empty_application_data
       end
     end
 
-    # TODO: remove remaining teachercon references
-    def get_optional_columns(_regional_partner_value)
-      {accepted_teachercon: false, registered_workshop: false}
+    def get_csv_text(applications, role)
+      prefetch applications, role: role
+      course = role[0..2] # course is the first 3 characters in role, e.g. 'csf'
```
Comment (mehalshah) [inline dashboard/app/controllers/api/v1/pd/applications_controller.rb]: Prefer splitting the string on the _ and taking the first part

File: dashboard/app/models/pd/application/facilitator1819_application.rb [hunk 2]
```diff
@@ -483,7 +483,6 @@ def self.csv_header(course, user)
       end
     end
 
-    # @override
     def self.cohort_csv_header(optional_columns)
```
Comment (mehalshah) [inline dashboard/app/models/pd/application/facilitator1819_application.rb:486]: Can we delete this function entirely then?

File: dashboard/app/models/pd/application/principal_approval1920_application.rb [hunk 3]
```diff
@@ -216,5 +216,38 @@ def additional_text_fields
...
+
+          option ||= OTHER_WITH_TEXT
+          additional_text_field_name ||= "#{field_name}_other".to_sym
+          hash[field_name] = self.class.answer_with_additional_text hash, field_name, option, additional_text_field_name
+          hash.delete additional_text_field_name
+        end
+      end
+    end
+
+    def underrepresented_minority_percent
```
Comment (mehalshah) [inline dashboard/app/models/pd/application/principal_approval1920_application.rb]: This is already implemented in the principal approval base class. Still need it here?

File: dashboard/app/models/pd/application/principal_approval1920_application.rb [hunk 4]
```diff
@@ -216,5 +216,38 @@ def additional_text_fields
...
+      percentages = sanitize_form_data_hash
+      percentages[:american_indian].to_i + percentages[:hispanic].to_i + percentages[:black].to_i + percentages[:pacific_islander].to_i
+    end
+
+    def school
+      @school ||= School.includes(:school_district).find_by(id: sanitize_form_data_hash[:school])
```
Comment (mehalshah) [inline dashboard/app/models/pd/application/principal_approval1920_application.rb:235]: Not sure why we're setting a member variable here

File: dashboard/app/models/pd/application/teacher1819_application.rb [hunk 5]
```diff
@@ -53,7 +53,6 @@ def teachercon_registration
       Pd::Teachercon1819Registration.find_by_pd_application_id(id)
     end
 
-    # @override
     def self.cohort_csv_header(optional_columns)
```
Comment (mehalshah) [inline dashboard/app/models/pd/application/teacher1819_application.rb:56]: Same here - do we still need these cohort_csv functions?

File: dashboard/app/models/school_stats_by_year.rb [hunk 6]
```diff
@@ -97,4 +97,13 @@ def has_high_school_grades?
   def has_k8_grades?
     grade_kg_offered || grade_01_offered || grade_02_offered || grade_03_offered || grade_04_offered || grade_05_offered || grade_06_offered || grade_07_offered || grade_08_offered
   end
+
+  def urm_percent
+    percent_of_students(student_am_count + student_hi_count + student_bl_count + student_hp_count)
+  end
+
+  # returns what percent "count" is of the total student enrollment
+  def percent_of_students(count)
```
Comment (mehalshah) [inline dashboard/app/models/school_stats_by_year.rb:106]: No need for this method if we're only calling it once

File: lib/cdo/shared_constants/pd/teacher1920_application_constants.rb [hunk 7]
```diff
@@ -196,5 +276,135 @@ def self.clean_multiline(string)
         :willing_to_travel
       ]
     }
+
+    SCHOLARSHIP_QUESTIONS = [
```
Comment (mehalshah) [inline lib/cdo/shared_constants/pd/teacher1920_application_constants.rb:282]: Something might be wrong with your merge - this should be removed

File: lib/cdo/shared_constants/pd/teacher1920_application_constants.rb [hunk 8]
```diff
@@ -129,6 +129,86 @@ def self.clean_multiline(string)
       }
     }.freeze
 
+    CSV_LABEL_OVERRIDES = {
```
Comment (mehalshah) [inline lib/cdo/shared_constants/pd/teacher1920_application_constants.rb]: A lot of these strings are identical to the strings above. These aren't overriding the strings, they are sometimes duplicating them.

I suggest flattening the above hash map, and keeping this as just the strings that differ from the above map. Can explain in more detail in person

**Comment by mehalshah** [comment]
If this test run passes, go ahead and merge in. Otherwise you might have to rebase off my merge changes

**Pull request merged**